questionary/prompts/select.py

```diff
@@ -23,6 +23,7 @@ def select(
     use_indicator: bool = False,
     use_pointer: bool = True,
     instruction: Text = None,
+    initial_choice: Optional[Union[Text, Choice, Dict[Text, Any]]] = None,
```

Comment:
IMHO this should be `default`, just like in any other question.

Reply:
Doing that would mean `default` behaves differently on the `select` type and `checkbox` type. As I said in #57 (where discussion is already ongoing), I think it is better we leave `default` alone on the `checkbox` type (seeing as it isn't broken there), and remove it completely from `select` (seeing as it doesn't do anything).

We can then add this new parameter to both `select` and `checkbox`, which indicates the initial pointed to item.

Reply:
> Doing that would mean `default` behaves differently on the `select` type and `checkbox` type

Yes, that'd be the expected behavior. You also don't expect it to work the same way in a text input, etc.

Honestly I see no practical use in adding `initial_choice` in checkboxes, the same as I don't see a reason to add it here: just making `default` work as expected should be enough. :roll_eyes:

Reply:
> Yes, that'd be the expected behavior. You also don't expect it to work the same way in a text input, etc.

No, however `select` and `checkbox` are similar and it would be confusing to have parameters with the same name, that do different things.

> Honestly I see no practical use in adding `initial_choice` in checkboxes, the same as I don't see a reason to add it here

Many of the use cases for being able to point to an arbitrary initial value for the `select` type are also applicable to the `checkbox` type. I see no reason to restrict functionality here.

Reply:
I don't think there is a use case where you'd want to point to any other but the default item in a select, is there? If not, I'd be for fixing the behaviour of default and avoid adding another attribute

Reply:
@tmbo How do you think it should behave with `checkbox` if we do this? Remove the ability to select the position?

Reply:
> The question is whether this enhancement should be added to `checkbox` too

I think the answer is no.

The fact you hit <kbd>Enter</kbd> in a checkbox just sends the checkbox answer, but it's going to end up with the same result, no matter where the cursor is located.

So the initial position in a checkbox is mostly irrelevant, and the initial position in a select must be exactly the same as the default value, always without exception.

So I agree with @tmbo. I think that all we need to do is to fix the behavior of `default` in selects.